pkg/apis/v1alpha2/serviceimport.go

```diff
@@ -0,0 +1,120 @@
+	Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
+type ServiceImportSpec struct{}
+
+// ServiceImportStatus describes derived state of an imported service.
+type ServiceImportStatus struct {
+	// +listType=atomic
+	Ports []v1.ServicePort `json:"ports"`
+	// ips will be used as the VIP(s) for this service when type is ClusterSetIP.
+	// +kubebuilder:validation:MaxItems:=1
+	// +optional
+	IPs []string `json:"ips,omitempty"`
+	// type defines the type of this service.
+	// Must be "ClusterSetIP" or "Headless".
+	// The "ClusterSetIP" type reflects exported Service(s) with type ClusterIP
+	// and the "Headless" type reflects exported Service(s) with type Headless.
+	// A ServiceImport with type ClusterSetIP SHOULD populate `.status.ips`
+	// with virtual IP address(es) where the service can be reached from within
+	// the importing cluster.
+	// If exported Services of the same name and namespace in a given ClusterSet
+	// have differing types, a "Conflict" status condition SHOULD be reported in
+	// ServiceExport status.
+	// +kubebuilder:validation:Enum=ClusterSetIP;Headless
+	Type ServiceImportType `json:"type"`
+	// sessionAffinity is used to maintain client IP based session affinity.
+	// Supports "ClientIP" and "None". Defaults to "None".
+	// Reflects the `.spec.sessionAffinity` configuration of the underlying
+	// exported Service. Ignored when exported Service type is Headless. If
+	// exported Services of the same name and namespace in a given ClusterSet have
+	// differing session affinity configuration, a "Conflict" status condition
+	// SHOULD be reported in ServiceExport status.
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	// +optional
+	SessionAffinity v1.ServiceAffinity `json:"sessionAffinity,omitempty"`
+	// sessionAffinityConfig contains session affinity configuration.
+	// Reflects the `.spec.sessionAffinityConfig` configuration of the underlying
+	// exported Service. Ignored when exported Service type is "Headless". If
+	// exported Services of the same name and namespace in a given ClusterSet have
+	// differing session affinity configuration, a "Conflict" status condition
+	// SHOULD be reported in ServiceExport status.
+	// +optional
+	SessionAffinityConfig *v1.SessionAffinityConfig `json:"sessionAffinityConfig,omitempty"`
+	// clusters is the list of exporting clusters from which this service
+	// was derived.
+	// +optional
+	// +patchStrategy=merge
+	// +patchMergeKey=cluster
+	// +listType=map
+	// +listMapKey=cluster
+	Clusters []ClusterStatus `json:"clusters,omitempty"`
+}
+
+// ClusterStatus contains service configuration mapped to a specific source cluster
+type ClusterStatus struct {
+	// cluster is the name of the exporting cluster. Must be a valid RFC-1123 DNS
+	// label.
+	Cluster string `json:"cluster"`
```

Comment:
We may want to centrally define this type as something more specific than `string` and add CEL validation to align with the suggested constraints in https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/2149-clusterid and https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/4322-cluster-inventory#cluster-name

Reply:
First pass at this in latest commit - I don't think we can capture the contextual constraints from KEP 2149 here, but it should at least be limited to RFC 1123 DNS labels now. I'm unsure if we can/should hoist this up somewhere to be reusable by e.g. ClusterProfile API, but this feels sufficient initially.